src/tlog-play.c

```diff
@@ -823,7 +823,12 @@ run(struct tlog_errs **perrs,
 
     /* Restore signal handlers */
     for (i = 0; i < TLOG_ARRAY_SIZE(exit_sig); i++) {
-        sigaction(exit_sig[i], NULL, &sa);
+        if(sigaction(exit_sig[i], NULL, &sa) == -1) {
+          grc = TLOG_GRC_ERRNO;
+          tlog_errs_pushc(perrs, grc);
+          tlog_errs_pushs(perrs, "Failed to retrieve an exit signal action");
+          goto cleanup;
```

Comment:
This shouldn't go to `cleanup`, it's already there. It should just add errors to the stack and get on with the cleanup.